plugins/admissions/models.py

```diff
@@ -0,0 +1,157 @@
+"""
+Models for the elCID admissions plugin
+"""
+from django.db import models
+from opal.models import Patient, PatientSubrecord
+
+from plugins.admissions import constants
+
+
+class PatientEncounterStatus(PatientSubrecord):
+    _is_singleton = True
+
+    has_encounters = models.BooleanField(default=False)
+
+
+class Encounter(models.Model):
+    """
+    An Encounter at the hospital as defined by Cerner / ADT
+
+    These are transmitted as HL7 messages and then stored in an RFH
+    SQL database which we poll to populate this table.
+
+    Here we replicate that database table closely, even though we don't
+    currently use all fields.
+
+    Our table should be the equivalent of the upstream table, filtered
+    to only inlcude our elCID cohort.
+    """
+    # We link the upstream data to our system via a single foreign key to the
+    # Patient object. In practice we take the MRN fields and match them
+    # with patient.demographics().hospital_number
+    patient = models.ForeignKey(Patient, on_delete=models.CASCADE, related_name='encounters')
+
+    # Note: Field - level comments here are from the documentation provided by the integration team
+    upstream_id                        = models.IntegerField(blank=True, null=True)              # Internal SQL Database identifier - Primary Key
+    msh_7                              = models.DateTimeField(blank=True, null=True)             # HL7 Message Datetime sent from Cerner
+    msh_9_msg_type                     = models.CharField(blank=True, null=True, max_length=4)   # Last HL7 Message update Type
```

Comment:
under the covers postgres stores the chars field in a text field anyway so including max_length can only burn us and not provide any benefit (not a blocker, but maybe we should just use text fields in future)